src/storage/ExtensionBasedMapper.ts

```diff
@@ -0,0 +1,129 @@
+import { posix } from 'path';
+import { types } from 'mime-types';
+import { RuntimeConfig } from '../init/RuntimeConfig';
+import { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { APPLICATION_OCTET_STREAM, TEXT_TURTLE } from '../util/ContentTypes';
+import { ConflictHttpError } from '../util/errors/ConflictHttpError';
+import { NotFoundHttpError } from '../util/errors/NotFoundHttpError';
+import { trimTrailingSlashes } from '../util/Util';
+import { FileIdentifierMapper } from './FileIdentifierMapper';
+
+const { join: joinPath, normalize: normalizePath } = posix;
+
+export interface PathAbstraction {
+
+  /**
+   * The parent directory path of the file.
```

Comment:
```suggestion
   * The absolute file path of the parent directory of the resource.
```

(It's absolute, right? If not, remove `absolute file`.)